### PR TITLE
Code of conduct

### PR DESCRIPTION
**Pull request by aradhakrishnanGFDL**
**Description**

This PR addresses having a code of conduct for the MDTF framework open-source project. It's community facing and provides general ground rules, expectations.

### CODE REVIEW

**Comment by aradhakrishnanGFDL** [comment]
All, this is a preliminary version of the code of conduct. Opening review from all mini-leads members offline before this can be merged in. Expecting to have an initial version merged in before October 2nd.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+   technical, happen all the time and this is no exception. 
+   People perceive their own behaviors very differently than others perceive them. 
+   What motivates one person may cause stress for another. 
+   A few steps worth remembering include the following:
+
+   * Other people do not see things the same way you do, 
+   do not feel the same as you, or interpret the world the way you do. 
+   Consider different people’s perspectives on things like change, 
+   drive, structure, process, big picture, and small details.
+   * Ask for feedback and listen to them.
```
Comment (bitterbark) [inline CODE_OF_CONDUCT.md:48]: Suggest changing this line to: Ask for and listen to feedback

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+   people of all backgrounds and identities. This includes, but is not limited
+   to, members of any race, ethnicity, culture, national origin, color,
+   immigration status, social and economic class, educational level, sex, sexual
+   orientation, gender identity and expression, age, physical appearance, family
+   status, technological or professional choices, academic
+   discipline, religion, mental ability, and physical ability.
+
+* **2. Be considerate and respectful**.    
+   Your work will be used by other people, and you in turn
+   will depend on the work of others. Any decision you take will affect users
```
Comment (bitterbark) [inline CODE_OF_CONDUCT.md:26]: suggest: any decision you *make*

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+   communicating with someone with a different primary language or cultural
+   background. Not all of us will agree all the time, but disagreement is
+   no excuse for poor behavior or poor manners.
+   Do not insult or put down other community members. A community where people
+   feel uncomfortable or threatened is not a productive one.    
+   
+   
+* **3. Consider Impact vs. Intent**.    
+   Disagreements, both social and
+   technical, happen all the time and this is no exception.
```
Comment (bitterbark) [inline CODE_OF_CONDUCT.md:39]: Not sure what "this is no exception" means here.

**Review by bitterbark — COMMENTED** [review]
Looks good, I made a few small suggestions (sorry I don't see how to edit/suggest text directly!)

**Comment by jkrasting** [comment]
Good idea @aradhakrishnanGFDL.

Maybe a section about "Failure to adhere to this policy will ..." ?

File: CODE_OF_CONDUCT.md (hunk 2 above)
Comment (bitterbark) [inline CODE_OF_CONDUCT.md:26]:
```suggestion
   will depend on the work of others. Any decision you make will affect users
```

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (bitterbark) [inline CODE_OF_CONDUCT.md:48]:
```suggestion
   * Ask for and listen to feedback. 
```

**Review by bitterbark — CHANGES_REQUESTED** [review]

File: CODE_OF_CONDUCT.md (hunk 3 above)
Comment (aradhakrishnanGFDL) [inline CODE_OF_CONDUCT.md:39]:
```suggestion
   technical, happen all the time and this project forum is no exception. 
```

Comment (aradhakrishnanGFDL) [inline CODE_OF_CONDUCT.md]:
```suggestion
   technical, happen all the time and this project community is no exception. 
```

**Comment by aradhakrishnanGFDL** [comment]
> Good idea @aradhakrishnanGFDL.
> 
> Maybe a section about "Failure to adhere to this policy will ..." ?

 @jkrasting, yes! we need the following section to be complete. This will need more input, just reaching out to entire mini-leads is not the right approach. Usually, few candidate members make a council or steering team (and a mailing list). General reporting guidelines through their respective organizations is okay, but not quite efficient as it spans outside the org. Not sure how to approach it for our work here. Perhaps, Dan can provide pointers here from NOAA CPO perspective- thoughts anyone? 

https://github.com/NOAA-GFDL/MDTF-diagnostics/pull/417/files/8e99517ef539b3245ded38b343e48c5ebcb8abca#diff-ffdbe3a1e7ee93cacfc080b6c635ccf3a8f6b0f00f2fb884f78c6b5f9dac8fd2R87

**Review by bitterbark — APPROVED** [review]

**Review by wrongkindofdoctor — APPROVED** [review]

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
+## Code of Conduct
+
+Model Diagnostics Task Force framework team is dedicated to providing a respectful and inclusive community for everyone. We provide this code of conduct as part of this commitment, following similar standards for other community development projects, as cited in the references. We do not tolerate harassment or bullying of any community member in any form. This applies equally to all community members regardless of their title, seniority, background, gender, etc., in all forums managed by the MDTF framework team. This includes the mailing lists, our GitHub project space, slack workspaces, in-person events, and any other forums created by the project team. In addition, violations of this code outside these spaces may affect a person's ability to participate within them. As part of the code of conduct policy, we will provide below ground rules for all forums (e.g meetings) created by the project team, supporting project initiatives.
```
Comment (wrongkindofdoctor) [inline CODE_OF_CONDUCT.md:3]:
```suggestion
The Model Diagnostics Task Force framework team is dedicated to providing a respectful and inclusive community for everyone. We provide this code of conduct as part of this commitment, following similar standards for other community development projects, as cited in the references. We do not tolerate harassment or bullying of any community member in any form. This applies equally to all community members regardless of their title, seniority, background, gender, etc., in all forums managed by the MDTF framework team. This includes the mailing lists, our GitHub project space, slack workspaces, in-person events, and any other forums created by the project team. In addition, violations of this code outside these spaces may affect a person's ability to participate within them. As part of the code of conduct policy, we will provide below ground rules for all forums (e.g meetings) created by the project team, supporting project initiatives. 
```

File: CODE_OF_CONDUCT.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+   * Violent threats or violent language directed against another person
+   * Discriminatory jokes and language
+   * Posting sexually explicit or violent material
+   * Posting (or threatening to post) other people's personally identifying
+     information ("doxing")
+   * Personal insults, especially those related to gender, 
+   gender identity and expression, sexual orientation, ability, 
+   physical appearance, body size, race, ethnicity, 
+   religion, socioeconomic status, caste or creed
+   * Unwelcome sexual attention, stalking;
```
Comment (wrongkindofdoctor) [inline CODE_OF_CONDUCT.md:81]:
```suggestion
   * Unwelcome sexual attention or stalking
```

**Review by wrongkindofdoctor — COMMENTED** [review]
@aradhakrishnanGFDL Thank you! The Code of Conduct is in great shape. I just a made couple of grammatical suggestions

**Review by jkrasting — APPROVED** [review]

**Pull request merged**